src/config/watcher.rs

```diff
@@ -90,23 +130,23 @@ pub fn spawn_thread<'a>(
     Ok(())
 }
 
-fn create_watcher(
-    config_paths: &[PathBuf],
-) -> Result<
-    (
-        RecommendedWatcher,
-        Receiver<Result<notify::Event, notify::Error>>,
-    ),
-    Error,
-> {
-    info!("Creating configuration file watcher.");
-    let (sender, receiver) = channel();
-    let mut watcher = recommended_watcher(sender)?;
-    add_paths(&mut watcher, config_paths)?;
-    Ok((watcher, receiver))
-}
-
-fn add_paths(watcher: &mut RecommendedWatcher, config_paths: &[PathBuf]) -> Result<(), Error> {
+// fn create_watcher(
+//     config_paths: &[PathBuf],
+// ) -> Result<
+//     (
+//         RecommendedWatcher,
+//         Receiver<Result<notify::Event, notify::Error>>,
+//     ),
+//     Error,
+// > {
+//     info!("Creating configuration file watcher.");
+//     let (sender, receiver) = channel();
+//     let mut watcher = recommended_watcher(sender)?;
+//     add_paths(&mut watcher, config_paths)?;
+//     Ok((watcher, receiver))
+// }
+
+fn add_paths(watcher: &mut Box<dyn Watcher>, config_paths: &[PathBuf]) -> Result<(), Error> {
```

Comment:
I think you could use
```suggestion
fn add_paths<T: Watcher>(watcher: &mut T, config_paths: &[PathBuf]) -> Result<(), Error> {
```